payment/controllers.go

```diff
@@ -30,6 +31,126 @@ func Router(service *Service) chi.Router {
 	return r
 }
 
+// KeyRouter handles management of keys
+func KeyRouter(service *Service) chi.Router {
+	r := chi.NewRouter()
+	if os.Getenv("ENV") != "local" {
+		r.Use(middleware.SimpleTokenAuthorizedOnly)
+	}
+
+	r.Method("GET", "/{merchantId}", middleware.InstrumentHandler("GetKeys", GetKeys(service)))
+	r.Method("POST", "/", middleware.InstrumentHandler("CreateKey", CreateKey(service)))
+	r.Method("DELETE", "/{id}", middleware.InstrumentHandler("DeleteKey", DeleteKey(service)))
+	return r
+}
+
+// CreateKeyRequest includes information needed to create an order
+type CreateKeyRequest struct {
+	Merchant string `json:"merchant" valid:"-"`
+}
+
+// DeleteKeyRequest includes information needed to create an order
+type DeleteKeyRequest struct {
+	DelaySeconds int `json:"DelaySeconds" valid:"-"`
+}
+
+// CreateKey is the handler for creating keys for a merchant
+func CreateKey(service *Service) handlers.AppHandler {
+	return handlers.AppHandler(func(w http.ResponseWriter, r *http.Request) *handlers.AppError {
+		var req CreateKeyRequest
+		err := requestutils.ReadJSON(r.Body, &req)
+		if err != nil {
+			// FIXME Ask Ben what he would like us to do here instead of wrapError
+			return handlers.WrapError(err, "Error in request body", http.StatusBadRequest)
+		}
+
+		_, err = govalidator.ValidateStruct(req)
+		if err != nil {
+			return handlers.WrapValidationError(err)
+		}
+
+		encrypted, nonce := GenerateSecret()
+		// var Key Key
+		key, err := service.datastore.CreateKey(req.Merchant, encrypted, nonce)
```

Comment:
I see nothing in here validating that the merchant is who they say they are... Can someone create keys for merchants that they are not?

Reply:
Realized this is being called by only a whitelisted app.  Disregard.

Reply:
I would consider this API internal, as no merchants should ever directly interact with this. 

The plan is that we'll eventually branch this off into it's own service.